templates/repo/issue/milestones.tmpl

```diff
@@ -6,53 +6,49 @@
 			{{template "repo/issue/navbar" .}}
 			{{if and (or .CanWriteIssues .CanWritePulls) (not .Repository.IsArchived)}}
 				<div class="ui right">
-					<a class="ui green button" href="{{$.Link}}/new">{{.locale.Tr "repo.milestones.new"}}</a>
+					<a class="ui small green button" href="{{$.Link}}/new">{{.locale.Tr "repo.milestones.new"}}</a>
 				</div>
 			{{end}}
 		</div>
 		<div class="ui divider"></div>
 		{{template "base/alert" .}}
 
-		<div class="ui three column stackable grid">
-			<div class="column">
-				<div class="ui compact tiny menu">
-					<a class="item{{if not .IsShowClosed}} active{{end}}" href="{{.RepoLink}}/milestones?state=open&q={{$.Keyword}}">
-						{{svg "octicon-milestone" 16 "gt-mr-3"}}
-						{{.locale.PrettyNumber .OpenCount}}&nbsp;{{.locale.Tr "repo.issues.open_title"}}
-					</a>
-					<a class="item{{if .IsShowClosed}} active{{end}}" href="{{.RepoLink}}/milestones?state=closed&q={{$.Keyword}}">
-						{{svg "octicon-check" 16 "gt-mr-3"}}
-						{{.locale.PrettyNumber .ClosedCount}}&nbsp;{{.locale.Tr "repo.issues.closed_title"}}
-					</a>
-				</div>
+		<div class="milestone-list-header gt-df gt-ac gt-gap-3 gt-fw">
```

Comment:
These "gt-df gt-ac gt-gap-3 gt-fw" seems quite wordy.

Since it has a class `milestone-list-header`, is it better to put the styles on the particular CSS class name?

Reply:
Hmm, 4 classes is not so bad yet, and I find it easier to modify them here than to search for the CSS styles.

Reply:
I can accept one or two, at most three, but 4 or more "gt-" classes start making things unclear.

And there are already a lot of "gt-" helpers on various "dropdown" components, I think we should resolve the problem from framework level.

Reply:
> I can accept one or two, at most three, but 4 or more "gt-" classes start making things unclear.

I still think it's pointless to limit number of classes to arbritrary numbers while also making is less maintainable by loosely coupling HTML to styles via CSS. Have you taken a look at pure tailwind sites? 10-15 classes is very common on those. Just take a look at GitHub's own HTML 😆

Reply:
Random example pulled from GH, shows 11 classes in total. I do think it's a bit excessive but not overly so.

```html
<div class="file-header d-flex flex-md-row flex-column flex-md-items-center file-header--expandable js-file-header js-skip-tagsearch  sticky-file-header js-position-sticky js-position-sticky-stacked">

Reply:
I guess if a element is repeated, it will result in fewer bytes down the wire if the styles are in CSS, so I guess I would consider it there for the sake of reduced transfer size.

Reply:
If these helpers are only use it once, that's fine.

But, I do not see why contributors should copy&paste these unclear code again and again.

If the copying&pasting works well, what's the value of designing frameworks? Why we extract duplicate code into functions ....

Reply:
Just have to keep in mind that people will forget to delete unused CSS which can't happen with the helper method. In simple cases like this with single unique class, it's not a problem, but overall I estimate that probably 25% or more of our CSS is unused and dead.

Reply:
Many helper classes are also dead, and even more difficult to find. I remove some styles and helper classes in almost every fine-tune PR.

So we should design a framework well, people do not need to add unnecessary CSS classes, then no class is needed to be removed.

Reply:
I feel like this is a pointless discussion. Restricting helper classes to arbritrary numbers like 3-4 is nonsense. These elements are single-use, so there is not even a transfer size benefit when moving to CSS.

We should rather be using our time to discuss more serious issues than this.

Reply:
Actually I see that I missed `/issues` and `/pulls` pages headers which can share CSS, so I'll move these rules to CSS for the purpose of sharing them across multiple templates.